ban/commands/municipality.py

```diff
@@ -0,0 +1,147 @@
+from ban.commands import command, reporter
+from ban.core import models
+from ban.core import versioning
+from . import helpers
+
+
+@command
+@helpers.session
+def merge(destination, sources=[], name='', label='', **kwargs):
+    """
+    Municipality merge command.
+    Steps:
+        - for each Municipality to be removed:
+            - redirect the Municipality by insee and id
+            - create a new Group with its name
+            - attach all HouseNumbers of this Municipality to this new Group
+            - attach sources Groups and sources PostCodes to destination
+            - for each PostCode of the Municipality:
+                - change the name and the complement (if the complement is null)
+            - remove the Municipality
+    """
+    if destination in sources:
+        helpers.abort('Destination in sources')
+    if not name or not name.split():
+        helpers.abort('Name should not be empty')
+    if not label or not label.split():
+        helpers.abort('Label should not be empty')
+    try:
+        destination = models.Municipality.get(
+            models.Municipality.insee == destination)
+    except models.Municipality.DoesNotExist:
+        helpers.abort('Destination does not exist')
+    if not sources:
+        helpers.abort('No sources')
+    sources_inst = []
+    # Make sure all sources exist before processing any of them.
+    for source in sources:
+        print(source)
+        try:
+            source = models.Municipality.get(
+                models.Municipality.insee == source)
+        except models.Municipality.DoesNotExist:
+            helpers.abort('Source {} does not exist'.format(source))
+        else:
+            sources_inst.append(source)
+    source_done = []
+    areas = []
+    db = models.Municipality._meta.database
+    with db.atomic():
+        process_postcode(destination, destination, label)
+        group_to_municipality(destination, destination, areas, label)
+        for source in sources_inst:
+            # Make sure that the source has not be treated yet
+            if source.insee not in source_done:
+                source_done.append(source.insee)
+                process_source(destination, source, areas, label)
+        validator = models.Municipality.validator(
+            instance=destination,
+            name=name,
+            version=destination.version+1,
```

Comment:
Spaces [around operators](https://www.python.org/dev/peps/pep-0008/#other-recommendations) (here and below)